subsys/bluetooth/controller/Kconfig

```diff
@@ -44,12 +44,17 @@ config BT_LL_SOFTDEVICE
 	select BT_CTLR_LE_PATH_LOSS_MONITORING_SUPPORT
 	select BT_CTLR_SUBRATING_SUPPORT
 	select BT_LL_SOFTDEVICE_HEADERS_INCLUDE
+	imply BT_CONN_TX_NOTIFY_WQ if SOC_FLASH_NRF_RADIO_SYNC_MPSL
 	depends on (SOC_SERIES_BSIM_NRFXX || SOC_SERIES_NRF52X || SOC_COMPATIBLE_NRF5340_CPUNET ||\
 			    SOC_NRF54H20_CPURAD || SOC_SERIES_NRF54LX)
 	depends on DT_HAS_NORDIC_BT_HCI_SDC_ENABLED
 	help
 	  Use SoftDevice Link Layer implementation.
 
+	  The option implies using a separate workqueue for connection TX notify
+	  processing. This is done to prevent timeouts in MPSL flash sync when
+	  flash operations are done from the system workqueue context.
```

Comment:
Are you doing blocking operations on the system workqueue? It is very much discouraged to block a shared resource.

Reply:
We should look into making the flash API async

Reply:
Actually, a lot of applications use system workqueue for flash operations (delegating settings store to system workq was quite common practice). Not sure if we can prevent users from doing it (we would need to e.g. add an execution context assertion to validate the requirement properly).